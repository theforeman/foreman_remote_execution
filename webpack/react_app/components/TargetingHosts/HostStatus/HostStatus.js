import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const HostStatus = (props) => {
  const { status } = props;

  switch (status) {
    case 'cancelled':
      return (<Fragment><span className="pficon pficon-warning-triangle-o"></span> cancelled</Fragment>);
    case 'N/A':
      return (<Fragment><span className="fa fa-question"></span> N/A</Fragment>);
    case 'running':
      return (<Fragment><span className="pficon pficon-running"></span> running</Fragment>);
    case 'planned':
      return (<Fragment><span className="pficon pficon-build"></span> running</Fragment>);
    case 'warning':
      return (<Fragment><span className="pficon pficon-error-circle-o"></span> failed</Fragment>);
    case 'error':
      return (<Fragment><span className="pficon pficon-error-circle-o"></span> failed</Fragment>);
    case 'success':
      return (<Fragment><span className="pficon pficon-ok"></span> success</Fragment>);
    default:
      return (<span>{status}</span>);
  }
};

HostStatus.propTypes = {
  status: PropTypes.string,
};

export default HostStatus;
