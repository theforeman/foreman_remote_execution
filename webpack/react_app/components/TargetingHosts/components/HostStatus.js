import React from 'react';
import { Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

const HostStatus = ({ status }) => {
  switch (status) {
    case 'cancelled':
      return (
        <div>
          <Icon type="pf" name="warning-triangle-o" /> {status}
        </div>
      );
    case 'N/A':
      return (
        <div>
          <Icon type="fa" name="question" /> {status}
        </div>
      );
    case 'running':
      return (
        <div>
          <Icon type="pf" name="running" /> {status}
        </div>
      );
    case 'planned':
      return (
        <div>
          <Icon type="pf" name="build" /> {status}
        </div>
      );
    case 'warning':
    case 'error':
      return (
        <div>
          <Icon type="pf" name="error-circle-o" /> {__('failed')}
        </div>
      );
    case 'success':
      return (
        <div>
          <Icon type="pf" name="ok" /> {status}
        </div>
      );
    default:
      return <span>{status}</span>;
  }
};

HostStatus.propTypes = {
  status: PropTypes.string.isRequired,
};

export default HostStatus;
