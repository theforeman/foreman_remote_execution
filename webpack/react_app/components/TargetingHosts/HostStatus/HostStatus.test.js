import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import HostStatus from './HostStatus';

const fixtures = {
  'renders HostStatus': {
    status: 'success',
  },

};
describe('HostStatus', () =>
  testComponentSnapshotsWithFixtures(HostStatus, fixtures));
