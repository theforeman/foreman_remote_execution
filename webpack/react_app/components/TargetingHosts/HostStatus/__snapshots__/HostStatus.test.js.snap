// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostStatus renders HostStatus 1`] = `
<Fragment>
  <span
    className="pficon pficon-ok"
  />
   success
</Fragment>
`;
